BE/src/main/java/louie/dong/airbnb/accommodation/AccommodationService.java

```diff
@@ -26,91 +27,63 @@
-
-		return new AccommodationDetailPriceResponse(accommodation.getPrice(), date, totalPrice,
-			WEEKLY.getDiscountRate(), discountPrice, accommodation.getCleaningFee(),
-			accommodation.getServiceFee(), accommodation.getAccommodationFee(), finalPrice);
-	}
-
-	public AccommodationSearchResponse findAccommodations(
-		AccommodationSearchRequest accommodationSearchRequest) {
+    private final AccommodationRepository accommodationRepository;
+    private final WishRepository wishRepository;
+
+    public AccommodationPriceResponse findPrices(String country) {
+        List<Integer> prices = accommodationRepository.findPricesByAccommodation(country);
+        return new AccommodationPriceResponse(calculateAverage(prices), prices);
+    }
+
+    public AccommodationDetailResponse findById(Long id) {
+        Accommodation accommodation = getAccommodationOrThrow(id);
+        boolean wish = wishRepository.existsByAccommodationId(id);
+        return new AccommodationDetailResponse(accommodation, wish);
+    }
+
+    public AccommodationDetailPriceResponse findDetailPrice(Long id,
+        AccommodationDetailPriceRequest accommodationDetailPriceRequest) {
+        Accommodation accommodation = getAccommodationOrThrow(id);
+        LocalDate checkIn = accommodationDetailPriceRequest.getCheckIn();
+        LocalDate checkOut = accommodationDetailPriceRequest.getCheckOut();
+        return new AccommodationDetailPriceResponse(checkIn, checkOut, accommodation);
+    }
+
+    public AccommodationSearchResponse findAccommodations(
+        AccommodationSearchRequest accommodationSearchRequest) {
 		LocalDate checkIn = accommodationSearchRequest.getCheckIn();
 		LocalDate checkOut = accommodationSearchRequest.getCheckOut();
 
-		List<Accommodation> accommodations = accommodationRepository.searchAccommodations(
-			accommodationSearchRequest.getCountry(),
-			LocalDateTime.of(checkIn, LocalTime.of(0, 0)),
-			LocalDateTime.of(checkOut, LocalTime.of(0, 0)),
-			accommodationSearchRequest.getMinPrice(), accommodationSearchRequest.getMaxPrice(),
-			accommodationSearchRequest.getGuestCount());
-
-		List<AccommodationResponse> accommodationResponses = createAccommodationResponses(
-			accommodations, checkIn, checkOut);
-		return new AccommodationSearchResponse(accommodations.size(), accommodationResponses);
-	}
+		List<Accommodation> accommodations = accommodationRepository.
+			searchAccommodations(accommodationSearchRequest);
 
-	private List<AccommodationResponse> createAccommodationResponses(
-		List<Accommodation> accommodations, LocalDate checkIn, LocalDate checkOut) {
-		int stayNight = (int) ChronoUnit.DAYS.between(checkIn, checkOut);
-		return accommodations.stream()
-			.map(accommodation -> new AccommodationResponse(accommodation,
-				stayNight * accommodation.getPrice(), accommodation.existsWish()))
-			.collect(Collectors.toList());
-	}
+        validAccommodations(accommodations);
```

Comment:
위에서 설명한 일급컬렉션을 사용했다면?
```suggestion
        Accommodations accommodations = accommodationRepository.searchAccommodations(accommodationSearchRequest);
        accommodations.validate();
```

Reply:
아래 코멘트 함께 참고
> validation 또한 도메인 객체 자신이 가지고 있어야 할 기능입니다.
객체 자신이 잘못된건 본인이 알고 있어야 해요, 왜 잘못됐는지를 도메인이 가지고 있어야 한다고 생각합니다 :)